File: python/delete_credentials.py
#!/usr/bin/env python3
"""Delete credential — query param id= (not path). Requires TRUNK_SID, CREDENTIAL_ID."""
from _client import delete
import os

trunk = os.environ.get("TRUNK_SID")
cid = os.environ.get("CREDENTIAL_ID")
if not trunk or not cid:
    print("Error: TRUNK_SID and CREDENTIAL_ID required")
    exit(1)

print("Deleting credential...")
delete(f"/trunks/{trunk}/credentials", query={"id": cid})

File: java/DeleteCredentials.java
public class DeleteCredentials {
    public static void main(String[] args) throws Exception {
        String trunk = System.getenv("TRUNK_SID");
        String cid = System.getenv("CREDENTIAL_ID");
        if (trunk == null || cid == null || trunk.isEmpty() || cid.isEmpty()) {
            System.err.println("Error: TRUNK_SID and CREDENTIAL_ID required");
            System.exit(1);
        }
        System.out.println("Deleting credential...");
        _Client.delete("/trunks/" + trunk + "/credentials", "id=" + java.net.URLEncoder.encode(cid, "UTF-8"));
    }
}
